src/prods/p10.py

```diff
@@ -0,0 +1,155 @@
+from prods.base import *
+
+class P10(Production):
+    def __init__(self) -> None:
+        self.lhs: Graph = self.__create_lhs()
+        self.rev_mapping: Dict[NodeHandle, NodeHandle] | None = None
+
+    def get_lhs(self) -> Graph:
+        return self.lhs
+
+    def reset(self):
+        self.rev_mapping = None
+
+    def __create_lhs(self) -> Graph:
+        graph = util.basic_pentagon(for_lhs=True)
+        # handle is 6 because the interior is 5
+        graph.add_hanging_node(u=0, v=1, handle=6)
+        return graph
+
+    def is_mapping_feasible(self, graph: Graph, mapping: Dict[NodeHandle, NodeHandle]) -> bool:
+        # Now we have mapping of lhs nodes to `graph` nodes.
+        self.rev_mapping = util.reverse_dict_mapping(mapping)
+
+        # Aliasing for convenience
+        rev_mapping = self.rev_mapping
+
+        # get square's vertices
+        vertices = [
+            graph.node_for_handle(rev_mapping[i]) for i in range(0, 5)
+        ]
+        # get inner node, the hyperedge
+        node_p = graph.node_for_handle(rev_mapping[5])
+        node_h = graph.node_for_handle(rev_mapping[6])
+
+        # check vertices' labels
+        for node in vertices:
+            if node.attrs.label != 'v' or node.attrs.hanging:
+                return False
+
+        # in case of inner node, the `hanging' parameter means that the
+        # polygon is chosen to be broken; it is the same as the R parameter 
+        if not node_p.attrs.hanging or node_p.attrs.label != 'p':
+            return False
+
+        if not node_h.attrs.hanging:
+            return False
+
+        # verify edges coming out of the inner node
+        for node in vertices:
+            edge = graph.edge_for_handles(node.handle, node_p.handle)
+            if edge.attrs.kind != 'p':
+                return False
+
+        idx_nodes = [1, 4, 2, 3, 0]
+        nodes_shuffled = [vertices[idx] for idx in idx_nodes]
+        for node_a, node_b in it.pairwise([node_h] + nodes_shuffled + [node_h]):
+            edge = graph.edge_for_handles(node_a.handle, node_b.handle)
+            if edge.attrs.kind != 'e':
+                return False
+
+        return True
```

Comment:
Hopefully, after #17 was merged we won't need to implement this method anymore 🥳 (I hope)